src/adapter/src/catalog.rs

```diff
@@ -6102,6 +6205,34 @@ impl From<ReplicaLocation> for SerializedReplicaLocation {
     }
 }
 
+/// A [`Role`] that is serialized as JSON and persisted to the catalog
+/// stash. This is a separate type to allow us to evolve the on-disk format
+/// independently from the SQL layer.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Ord, PartialOrd)]
+pub struct SerializedRole {
+    pub name: String,
+    // TODO(jkosh44): Remove in v0.46.0
+    #[serde(default)]
```

Comment:
This is a bit spooky. I'd prefer to make this struct exactly what we need going forward, and making a copy of this with an accurate data model right inside the migration closure. That can then use an `Option` instead of rust defaults, and we'll then not need to worry about remembering to remove the serde derive.

Reply:
I'm having trouble understanding this approach, can we chat about it today? As far as I know, the values are deserialized based on the structs inside the `TypedCollection`s. Is there a way to pass a new struct to a stash transaction to use for deserialization?

Reply:
Ok, I updated this to reflect what we talked about to use `Option`.